### PR TITLE
Make debugging matchedRule.rule optional to remove error

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Reviewer:**

## Description:

Fixes:
https://app.asana.com/0/892838074342800/1200955361390255/f

Also untested as working on something else.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/devtools.es6.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ function connected (port) {
             const matchedTracker = trackers.getTrackerData(requestData.url, siteUrl, requestData)
             if (tracker.matchedRule) {
                 // find the rule for this url
-                const ruleIndex = matchedTracker.tracker.rules.findIndex((r) => r.rule.toString() === tracker.matchedRule)
+                const ruleIndex = matchedTracker.tracker.rules.findIndex((r) => r?.rule?.toString() === tracker.matchedRule)
```
Comment (kdzwinel) [inline shared/js/background/devtools.es6.js:27]: rule itself can be null? This sounds like an error in data, should we hide it here?

Comment (jonathanKingston) [reply shared/js/background/devtools.es6.js:27]: r shouldn't be no, will remove the first ?

Comment (kdzwinel) [reply shared/js/background/devtools.es6.js:27]: You haven't removed the first `?` :)

**Review by kdzwinel — APPROVED** [review]
LGTM 👍

**Pull request merged**